src/fmeval/data_loaders/data_sources.py

```diff
@@ -110,14 +114,25 @@ def __reduce__(self):
 
 def get_s3_client(uri: str) -> boto3.client:
     """
-    Util method to return boto3 s3 client. For built-in datasets, the boto3 client region is default to us-west-2 as
-        the bucket is not accessible in opt-in regions.
+    Util method to return boto3 s3 client. For built-in datasets, the boto3 client region is default to us-west-2 for
+        commercial regions as the bucket is not accessible in opt-in regions.
+        For us-isof partition, built-in datasets are located in us-isof-south-1 region.
+
     :param uri: s3 dataset uri
     :return: boto3 s3 client
     """
-    s3_client = (
-        boto3.client("s3", region_name=BUILT_IN_DATASET_DEFAULT_REGION)
-        if uri.startswith(BUILT_IN_DATASET_PREFIX)
-        else boto3.client("s3")
-    )
+    session = boto3.session.Session()
+    region = session.region_name
+    if region in BUILT_IN_DATASET_ISO_REGIONS.keys():
```

Comment:
Question: if later we want to support other ISO regions, are we going to keep updating BUILT_IN_DATASET_ISO_REGIONS?
I'd prefer use prefix check so it will be automatically expand to other regions.

Reply:
For new iso regions in other iso partitions we'll need to add the built-in datasets to a region in that partition and update the `BUILD_IN_DATASET_ISO_REGIONS` map, since iso partitions (eg. iso-f and iso-e) can't access each other.